env/src/main/java/com/sap/cloud/security/config/k8s/K8sServiceConfigurationResolver.java

```diff
@@ -0,0 +1,157 @@
+
+/**
+ * The K8sServiceConfigurationResolver manages the service secret access from
+ * the file system.
+ */
+class K8sServiceConfigurationResolver {
+	private static final Logger LOGGER = LoggerFactory.getLogger(K8sServiceConfigurationResolver.class);
+
+	private String xsuaaPath;
+	private String iasPath;
+	private String serviceManagerPath;
+
+	K8sServiceConfigurationResolver() {
+		resolveServiceConfigurationPaths();
+	}
+
+	/**
+	 * Loads Service manager OAuth2 service configuration.
+	 *
+	 * @return OAuth2 service configuration
+	 */
+	@Nullable
+	OAuth2ServiceConfiguration loadServiceManagerConfig() {
+		File[] serviceBindings = new File(serviceManagerPath).listFiles();
+		if (serviceBindings == null || serviceBindings.length == 0) {
+			LOGGER.warn("No service-manager binding was found in {}", serviceManagerPath);
+			return null;
+		}
+		Map<String, String> smPropertiesMap = getServiceProperties(serviceBindings[0]);
+		return OAuth2ServiceConfigurationBuilder.forService(Service.XSUAA).withProperties(smPropertiesMap).build();
+	}
+
+	/**
+	 * Loads XSUAA or IAS OAuth2 service configuration.
+	 *
+	 * @param service
+	 *            IAS or XSUAA service
+	 * @return the map of service instance name and it's OAuth2 service
+	 *         configurations
+	 */
+	Map<String, OAuth2ServiceConfiguration> loadOauth2ServiceConfig(Service service) {
+		Map<String, OAuth2ServiceConfiguration> allServices = new HashMap<>();
+		File[] serviceBindings = getServiceBindings(service);
+		if (serviceBindings == null || serviceBindings.length == 0) {
+			LOGGER.warn("No service bindings for {} service were found", service);
+		} else {
+			LOGGER.debug("Found {} {} service bindings", serviceBindings.length, service);
+			for (File binding : serviceBindings) {
+				Map<String, String> servicePropertiesMap = getServiceProperties(binding);
+				if (!servicePropertiesMap.isEmpty()) {
+					OAuth2ServiceConfiguration config = OAuth2ServiceConfigurationBuilder.forService(service)
+							.withProperties(servicePropertiesMap)
+							.build();
+					allServices.put(binding.getName(), config);
+				}
+			}
+		}
+		return allServices;
+	}
+
+	@Nullable
+	private File[] getServiceBindings(Service service) {
+		String path = service == Service.XSUAA ? xsuaaPath : iasPath;
+		LOGGER.debug("Retrieving {} service bindings from K8s secret file {}", service, path);
+		return new File(path).listFiles();
+	}
+
+	private List<File> getBindingFiles(@Nonnull File binding) {
+		File[] bindingFiles = new File(binding.getPath()).listFiles();
+		if (bindingFiles == null || bindingFiles.length == 0) {
+			LOGGER.warn("No service binding files were found for {}", binding.getName());
+			return Collections.emptyList();
+		}
+		return Arrays.stream(bindingFiles).filter(File::isFile)
+				.collect(Collectors.toList());
+	}
+
+	private Map<String, String> getServiceProperties(File binding) {
+		List<File> serviceBindingFiles = getBindingFiles(binding);
```

Comment:
this may change if we decide to support BTP service operator only (instead of bindings created via service catalog)
https://github.com/SAP/sap-btp-service-operator/blob/main/README.md

Reply:
interesting 👍🏻